src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs

```diff
@@ -208,65 +216,79 @@ public ActionResult DownloadExportFile([FromRoute] string fileName)
-                                                         .Where(x => x.Version.IsCompatibleWith(PlatformVersion.CurrentVersion))
-                                                         .GroupBy(x => x.Name)
-                                                         .Select(x => x.OrderByDescending(y => y.Version).First().Data)
-                                                         .ToList();
-                        //Convert relative  sample data urls to absolute
-                        foreach (var sampleDataInfo in sampleDataInfos)
-                        {
-                            if (!Uri.IsWellFormedUriString(sampleDataInfo.Url, UriKind.Absolute))
-                            {
-                                var uri = new Uri(sampleDataUrl);
-                                sampleDataInfo.Url = new Uri(uri, uri.AbsolutePath + "/" + sampleDataInfo.Url).ToString();
-                            }
-                        }
-                        retVal.AddRange(sampleDataInfos);
+                    new SampleDataInfo { Url = sampleDataUrl }
+                };
+            }
 
-                    }
-                }
-                else
+            //Discovery mode
+            var manifestUrl = sampleDataUrl + "\\manifest.json";
+            using (var client = new WebClient())
+            using (var stream = client.OpenRead(new Uri(manifestUrl)))
+            {
+                //Add empty template
+                var retVal = new List<SampleDataInfo>
                 {
-                    //Direct file mode
-                    retVal.Add(new SampleDataInfo { Url = sampleDataUrl });
+                    new SampleDataInfo { Name = "Empty" }
+                };
+
+                //Need filter unsupported versions and take one most new sample data
+                var sampleDataInfos = stream.DeserializeJson<List<SampleDataInfo>>()
+                    .Select(x => new
+                    {
+                        Version = SemanticVersion.Parse(x.PlatformVersion),
+                        x.Name,
+                        Data = x
+                    })
+                    .Where(x => x.Version.IsCompatibleWith(PlatformVersion.CurrentVersion))
+                    .GroupBy(x => x.Name)
+                    .Select(x => x.OrderByDescending(y => y.Version).First().Data)
+                    .ToList();
+
+                //Convert relative  sample data urls to absolute
+                foreach (var sampleDataInfo in sampleDataInfos)
+                {
+                    if (!Uri.IsWellFormedUriString(sampleDataInfo.Url, UriKind.Absolute))
+                    {
+                        var uri = new Uri(sampleDataUrl);
+                        sampleDataInfo.Url = new Uri(uri, uri.AbsolutePath + "/" + sampleDataInfo.Url).ToString();
+                    }
                 }
+
+                retVal.AddRange(sampleDataInfos);
+
+                return retVal;
```

Comment:
In case you've refactored this method, please also rename `retVal` to `result`.
Also seems return row should not be in the scope of the using - do not see the necessity in that change.

Reply:
Renamed. About variable, we don't need it inside of scope, so we don't need to take out it.